### PR TITLE
Fix Scala deprecations as much as possible

### PR DESCRIPTION
**Pull request by exoego**
Fixed Scala standard library deprecations as much as possible.
Some deprecations remain to support Scala 2.12.

### CODE REVIEW

File: airframe-http-codegen/src/main/scala/wvlet/airframe/http/openapi/OpenAPI.scala [hunk 1]
```diff
@@ -158,7 +158,7 @@ object OpenAPI {
       additionalProperties: Option[SchemaOrRef] = None,
       items: Option[SchemaOrRef] = None,
       nullable: Option[Boolean] = None,
-      enum: Option[Seq[String]] = None
+      `enum`: Option[Seq[String]] = None
```
Comment (exoego) [inline airframe-http-codegen/src/main/scala/wvlet/airframe/http/openapi/OpenAPI.scala:161]: `enum` became keyword from Scala 3.

File: airframe-http-recorder/src/test/scala/wvlet/airframe/http/recorder/HttpRecorderTest.scala [hunk 2]
```diff
@@ -71,7 +71,7 @@ class HttpRecorderTest extends AirSpec {
     response.status shouldBe replayResponse.status
     replayResponse.headerMap.get("X-Airframe-Record-Time") shouldBe defined
     orderInsensitveHash(response.headerMap.toMap) shouldBe orderInsensitveHash(
-      (replayResponse.headerMap - "X-Airframe-Record-Time").toMap
+      replayResponse.headerMap.toMap - "X-Airframe-Record-Time"
```
Comment (exoego) [inline airframe-http-recorder/src/test/scala/wvlet/airframe/http/recorder/HttpRecorderTest.scala:74]: `-` for mutable map deprecated.

New code: Copying entries into new immutable map, then remove.

File: airframe-json/src/main/scala/wvlet/airframe/json/JSON.scala [hunk 3]
```diff
@@ -119,7 +119,7 @@ object JSON extends LogSupport {
     def toJSON: String
   }
 
-  final case object JSONNull extends JSONValue {
+  case object JSONNull extends JSONValue {
```
Comment (exoego) [inline airframe-json/src/main/scala/wvlet/airframe/json/JSON.scala:122]: This is not a deprecation, but a warning. `case object` is implicitly `final`, so it is redundant.

File: airframe-metrics/.jvm/src/main/scala/wvlet/airframe/metrics/TimeVector.scala [hunk 4]
```diff
@@ -48,7 +48,7 @@ case class TimeVector(x: Long, offset: Long, unit: TimeWindowUnit) {
 }
 
 object TimeVector {
-  private val durationPattern = "^([+-]|last|next)?([0-9]+)(s|m|d|h|w|M|q|y)".r("prefix", "num", "unit")
+  private val durationPattern = "^(?<prefix>[+-]|last|next)?(?<num>[0-9]+)(?<unit>s|m|d|h|w|M|q|y)".r
```
Comment (exoego) [inline airframe-metrics/.jvm/src/main/scala/wvlet/airframe/metrics/TimeVector.scala:51]: Naming group `(...)` in `.r("name")` is deprecated.
Instead, `(?<name>...)` is recommended.

File: airframe-rx-html/.js/src/test/scala/wvlet/airframe/http/rx/html/NestedElementTest.scala [hunk 5]
```diff
@@ -34,9 +34,7 @@ object NestedElementTest extends AirSpec {
     div(
       id -> "nested",
       // Seq[RxElement]
-      lst.toSeq.map { i => new MyElement(s"button ${i}") },
-      // Iterable[RxElement]
-      lst.toIterable.map { i => new MyElement(s"button ${i}") }
+      lst.toSeq.map { i => new MyElement(s"button ${i}") }
```
Comment (exoego) [inline airframe-rx-html/.js/src/test/scala/wvlet/airframe/http/rx/html/NestedElementTest.scala:37]: `toIterable` is deprecated, but there is no direct counterpart, so just deleted.

File: airframe-http-recorder/src/test/scala/wvlet/airframe/http/recorder/HttpRecorderTest.scala [hunk 6]
```diff
@@ -71,7 +71,7 @@ class HttpRecorderTest extends AirSpec {
     response.status shouldBe replayResponse.status
     replayResponse.headerMap.get("X-Airframe-Record-Time") shouldBe defined
     orderInsensitveHash(response.headerMap.toMap) shouldBe orderInsensitveHash(
-      (replayResponse.headerMap - "X-Airframe-Record-Time").toMap
+      (replayResponse.headerMap -= "X-Airframe-Record-Time").toMap
```
Comment (exoego) [inline airframe-http-recorder/src/test/scala/wvlet/airframe/http/recorder/HttpRecorderTest.scala:74]: `- key ` ... `clone() -= key` ... is deprecated, but `-=` is OK.
`-=` is a mutator method which is not compatible with `-`, but for this test it seems OK.

File: airframe-control/.jvm/src/main/scala/wvlet/airframe/control/Shell.scala [hunk 7]
```diff
@@ -65,6 +65,7 @@ object Shell extends LogSupport {
 
     // retrieve child processes
     val pb = prepareProcessBuilder(s"ps -o pid -p $pid | sed 1d", inheritIO = true)
+    // TODO: Use lazyLines when Scala 2.12 dropped
```
Comment (xerial) [inline airframe-control/.jvm/src/main/scala/wvlet/airframe/control/Shell.scala:68]: This will be when sbt supports Scala 3 😅

**Comment by xerial** [comment]
👍

**Pull request merged**